src/sage/rings/polynomial/laurent_polynomial_ring_base.py

```diff
@@ -488,16 +488,150 @@ def krull_dimension(self):
         """
         raise NotImplementedError
 
-    def random_element(self, low_degree=-2, high_degree=2, terms=5, choose_degree=False,*args, **kwds):
+    def random_element(self, min_valuation=-2, max_degree=2, *args, **kwds):
         """
+        Return a random polynomial with degree at most ``max_degree`` and
+        lowest valuation at least ``min_valuation``.
+
+        Uses the random sampling from the base polynomial ring then divides out
+        by a monomial to ensure correct ``max_degree`` and ``min_valuation``.
+
+        INPUT:
+
+        - ``min_valuation`` -- integer (default: ``-2``); the
+          minimal allowed valuation of the polynomial
+
+        - ``max_degree`` -- integer (default: ``2``); the
+          maximal allowed degree of the polynomial
+
+        - ``*args, **kwds`` -- passed to the random element generator of the
+          base polynomial ring and base ring itself
+
         EXAMPLES::
 
-            sage: LaurentPolynomialRing(QQ, 2, 'x').random_element()
+            sage: L.<x> = LaurentPolynomialRing(QQ)
+            sage: f = L.random_element()
+            sage: f.degree() <= 2
+            True
+            sage: f.valuation() >= -2
+            True
+            sage: f.parent() is L
+            True
```

Comment:
Can you add the same test but with `L.<x> = LaurentPolynomialRing(QQ, 1)` (this won't have the `shift` method for the elements)?

Reply:
Yes, but I moved it to TEST rather than examples as I don't think it's good to encourage the construction of this weird "multivariate" polynomial in one generator. Is this OK? Also, this case was previously handled with my randomised testing.